src/job_queue.jl

```diff
@@ -1,23 +1,67 @@
-using AWSSDK.Batch: describe_job_queues
-
 struct JobQueue
     arn::String
 
-    function JobQueue(queue::AbstractString)
-        arn = job_queue_arn(queue)
+    function JobQueue(queue::AbstractString; aws_config::AbstractAWSConfig=global_aws_config())
+        arn = job_queue_arn(queue; aws_config)
         arn === nothing && error("No job queue ARN found for: $queue")
         new(arn)
     end
 end
 
 Base.:(==)(a::JobQueue, b::JobQueue) = a.arn == b.arn
-describe(queue::JobQueue) = describe_job_queue(queue)
-describe_job_queue(queue::JobQueue) = describe_job_queue(queue.arn)
-max_vcpus(queue::JobQueue) = sum(max_vcpus(ce) for ce in compute_environments(queue))
+function describe(queue::JobQueue; aws_config::AbstractAWSConfig=global_aws_config())
+    describe_job_queue(queue; aws_config)
+end
+function describe_job_queue(queue::JobQueue; aws_config::AbstractAWSConfig=global_aws_config())
+    describe_job_queue(queue.arn; aws_config)
+end
+function max_vcpus(queue::JobQueue; aws_config::AbstractAWSConfig=global_aws_config())
+    sum(max_vcpus(ce; aws_config) for ce in compute_environments(queue; aws_config))
+end
+
+function _create_compute_environment_order(envs)
+    map(enumerate(envs)) do (i, env)
+        Dict{String,Any}("computeEnvironment"=>env, "order"=>i)
+    end
+end
+
+"""
+    create_job_queue(name, envs, priority=1; aws_config=global_aws_config())
+
+Create a job queue with name `name` and priority `priority` returning the associated `JobQueue` object.
+`envs` must be an iterator of compute environments given by ARN.
+
+See the AWS docs [here](https://docs.aws.amazon.com/batch/latest/APIReference/API_CreateJobQueue.html).
+"""
+function create_job_queue(name::AbstractString, envs, priority::Integer=1;
+                          enabled::Bool=true,
+                          tags::AbstractDict=Dict{String,Any}(),
+                          aws_config::AbstractAWSConfig=global_aws_config())
```

Comment:
Should also specify a type for `envs` here as well.

```suggestion
function create_job_queue(
    name::AbstractString, 
    envs, 
    priority::Integer=1;
    enabled::Bool=true,
    tags::AbstractDict=Dict{String,Any}(),
    aws_config::AbstractAWSConfig=global_aws_config()
)
```